lib/Doctrine/DBAL/Migrations/FileQueryWriter.php

```diff
@@ -4,51 +4,102 @@
 
 namespace Doctrine\DBAL\Migrations;
 
-/**
- * @since  1.6.0
- * @author Luís Cobucci <lcobucci@gmail.com>
- */
+use InvalidArgumentException;
+use function date;
+use function file_put_contents;
+use function is_dir;
+use function sprintf;
+
 final class FileQueryWriter implements QueryWriter
 {
-    /**
-     * @var string
-     */
+    /** @var string */
     private $columnName;
 
-    /**
-     * @var string
-     */
+    /** @var string */
     private $tableName;
 
-    /**
-     * @var null|OutputWriter
-     */
+    /** @var null|OutputWriter */
     private $outputWriter;
 
-    public function __construct(string $columnName, string $tableName, ?OutputWriter $outputWriter)
-    {
+    public function __construct(
+        string $columnName,
+        string $tableName,
+        ?OutputWriter $outputWriter
+    ) {
         $this->columnName   = $columnName;
         $this->tableName    = $tableName;
         $this->outputWriter = $outputWriter;
     }
 
     /**
-     * TODO: move SqlFileWriter's behaviour to this class - and kill it with fire (on the next major release)
-     * @param string $path
-     * @param string $direction
-     * @param array $queriesByVersion
-     * @return bool
+     * @param mixed[] $queriesByVersion
      */
-    public function write(string $path, string $direction, array $queriesByVersion) : bool
+    public function write(
+        string $path,
+        string $direction,
+        array $queriesByVersion
+    ) : bool {
+        $path   = $this->buildMigrationFilePath($path);
+        $string = $this->buildMigrationFile($queriesByVersion, $direction);
+
+        if ($this->outputWriter) {
+            $this->outputWriter->write(
+                "\n" . sprintf('Writing migration file to "<info>%s</info>"', $path)
+            );
+        }
+
+        return file_put_contents($path, $string) ? true : false;
+    }
+
+    /** @param string[][] $queriesByVersion */
+    private function buildMigrationFile(
+        array $queriesByVersion,
+        string $direction
+    ) : string {
+        $string = sprintf("-- Doctrine Migration File Generated on %s\n", date('Y-m-d H:i:s'));
+
+        foreach ($queriesByVersion as $version => $queries) {
+            $string .= "\n-- Version " . $version . "\n";
+
+            foreach ($queries as $query) {
+                $string .= $query . ";\n";
+            }
+
+            $string .= $this->getVersionUpdateQuery($version, $direction);
+        }
+
+        return $string;
+    }
+
+    private function getVersionUpdateQuery($version, string $direction) : string
     {
-        $writer = new SqlFileWriter(
-            $this->columnName,
+        if ($direction === Version::DIRECTION_DOWN) {
```

Comment:
Don't like this inlined here... Would be better extracted to some strategies like DownMigrator + UpMigrator.